mcs/class/System.Net.Http/System.Net.Http/HttpClient.wasm.cs

```diff
@@ -18,13 +18,34 @@ static HttpMessageHandler GetDefaultHandler()
         {
 
             if (GetHttpMessageHandler == null)
-                return GetFallback("No custom HttpClientHandler registered");
-
-            var handler = GetHttpMessageHandler();
-            if (handler == null)
-                return GetFallback($"Custom HttpMessageHandler is not valid");
-
-            return handler;
+            {
+                Type type = Type.GetType("WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler, WebAssembly.Net.Http");
+                if (type == null)
+                    return GetFallback ("Invalid WebAssembly Module? Cannot find WebAssembly.Net.Http.HttpClient.WasmHttpMessageHandler");
+
+                MethodInfo method = type.GetMethod("GetHttpMessageHandler", BindingFlags.Static | BindingFlags.NonPublic);
+                if (method == null)
+                    return GetFallback ("Your WebAssembly version does not support obtaining of the custom HttpClientHandler");
+
+                object ret = method.Invoke(null, null);
+                if (ret == null)
+                    return GetFallback ("WebAssembly returned no custom HttpClientHandler");
+
+                var handler = ret as HttpMessageHandler;
+                if (handler == null)
+                    return GetFallback ($"{ret?.GetType()} is not a valid HttpMessageHandler");
+                    
+                return handler;
+
+            }
+            else
```

Comment:
The class allows developers to supply their own Handler via `private static Func<HttpMessageHandler> GetHttpMessageHandler;`  we need to respect that so that is why there is an `else`.  The only time we will try to create a default is if there is not a helper already registered.  This was provided by https://github.com/mono/mono/pull/10152